src/jdk.jcmd/share/classes/sun/tools/jmap/JMap.java

```diff
@@ -227,7 +227,7 @@ private static void dump(String pid, String options)
                 // ignore format (not needed at this time)
             } else if (subopt.startsWith("gz=")) {
                compress_level = subopt.substring("gz=".length());
-               if (compress_level == null) {
+               if (compress_level == null || compress_level.length() == 0) {
```

Comment:
I don't think `substring` can return `null`, so the proper fix was to remove the null check and replace with your check for `== 0`.